### PR TITLE
Enable stateful RNN with cntk backend

### PR DESCRIPTION
**Pull request by souptc**
This pull request is for the stateful RNN support in cntk backend, which is missing in the first release.
Also enable the stateful rnn test on cntk backend.

### CODE REVIEW

File: keras/backend/cntk_backend.py [hunk 1]
```diff
@@ -2053,6 +2061,44 @@ def backward(self, state, root_gradients):
                 (num_old_batch,) + self.from_shape))
 
 
+class ConvertToBatch(C.ops.functions.UserFunction):
```
Comment (fchollet) [inline keras/backend/cntk_backend.py]: Please add a docstring.

File: keras/backend/cntk_backend.py [hunk 2]
```diff
@@ -2053,6 +2061,44 @@ def backward(self, state, root_gradients):
...
+                [batch_axis])]
+
+    def forward(self, arguments, device=None, outputs_to_retain=None):
+        return None, C.cntk_py.Value(arguments.data())
+
+    def backward(self, state, root_gradients):
+        return C.cntk_py.Value(root_gradients.data())
+
+
+class ConvertToStatic(C.ops.functions.UserFunction):
```
Comment (fchollet) [inline keras/backend/cntk_backend.py]: Please add a docstring.

**Comment by souptc** [comment]
Has add the doc string

File: keras/backend/cntk_backend.py (hunk 1 above)
Comment (fchollet) [inline keras/backend/cntk_backend.py]: In both docstrings, include all arguments to `__init__`.

**Pull request merged**